packages/react-devtools-extensions/src/generateHookMap.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import traverse, {Node, NodePath} from '@babel/traverse';
+import {string} from 'yargs';
+import {getHookNamesFromAST} from './astUtils';
+import {encode, decode} from 'sourcemap-codec';
+
+export type HookMap = {|
+  names: $ReadOnlyArray<string>,
+  mappings: $ReadOnlyArray<
+    $ReadOnlyArray<[number, number, number, number, number]>,
+  >,
+|};
+
+export type EncodedHookMap = {|
+  names: $ReadOnlyArray<string>,
+  mappings: string,
+|};
+
+export function generateHookMap(sourceAST: Node): HookMap {
```

Comment:
`File`?

```suggestion
export function generateHookMap(sourceAST: File): HookMap {
```

Reply:
ah, thanks! i've had a bit of trouble getting proper Flow coverage for this so wasn't sure what to use.